extensions/aws/processors/FetchS3Object.cpp

```diff
@@ -74,7 +74,7 @@ std::optional<aws::s3::GetObjectRequestParameters> FetchS3Object::buildFetchS3Re
     const std::shared_ptr<core::ProcessContext> &context,
     const std::shared_ptr<core::FlowFile> &flow_file,
     const CommonProperties &common_properties) const {
-  minifi::aws::s3::GetObjectRequestParameters get_object_params(common_properties.credentials, client_config_);
+  minifi::aws::s3::GetObjectRequestParameters get_object_params(common_properties.credentials, gsl::make_not_null(client_config_.get()));
```

Comment:
I would add a precondition about the validity of `client_config_` in each member that depends on it, just to explicitly document the requirements.

Reply:
Added checks in b3a7f4fb48c7e1ffa069c2bf9d2bc6ad01191b33